src/runtimes/docker/clients/DockerClientBase/DockerClientBase.ts

```diff
@@ -651,6 +654,42 @@ export abstract class DockerClientBase extends ConfigurableClient implements ICo
 
     //#endregion
 
+    //#region ImagesGenerateSbom Command
+
+    /**
+     * Generate the command line arguments for invoking a pull image command on
+     * a Docker-like client
+     * @param options Pull image command options
+     * @returns Command line arguments for pulling a container image
+     */
+    protected getGenerateSbomCommandArgs(options: ImageGenerateSbomCommandOptions): CommandLineArgs {
+        return composeArgs(
+            withArg('sbom'),
+            withNamedArg('--format', 'spdx-json', {shouldQuote: false}),
+            withArg(options.imageRef),
+        )();
+    }
+
+    protected async parseImageGenerateSbomCommandOutput(
+        options: ImageGenerateSbomCommandOptions,
+        output: string,
+        strict: boolean,
+    ): Promise<Array<ImageSbomItem>> {
+        const sboms = new Array<ImageSbomItem>();
+        sboms.push({raw: output});
+        return sboms;
+    }
+
+    async imageGenerateSbom(options: ImageGenerateSbomCommandOptions): Promise<PromiseCommandResponse<Array<ImageSbomItem>>> {
+        return {
+            command: this.commandName,
+            args: this.getGenerateSbomCommandArgs(options),
+            parse: (output, strict) => this.parseImageGenerateSbomCommandOutput(options, output, strict),

```

Comment:
Consider creating the return Array in place and eliminating the `parseImageGenerateSbomCommandOutput`